tests/std/tests/P1425R4_queue_stack_constructors/test.cpp

```diff
@@ -0,0 +1,200 @@
+
+    int result;
+#if _HAS_CXX23
+    stack s1(range.begin(), range.end());
+    static_assert(is_same_v<decltype(s1), stack<int, deque<int, allocator<int>>>>);
+    assert(s1.size() == size(some_data));
+    result = 5;
+    while (!s1.empty()) {
+        assert(s1.top() == result--);
+        s1.pop();
+    }
+
+    stack s2(range.begin(), range.end(), custom_allocator<int>{});
+    static_assert(is_same_v<decltype(s2), stack<int, deque<int, custom_allocator<int>>>>);
+    assert(s2.size() == size(some_data));
+    result = 5;
+    while (!s2.empty()) {
+        assert(s2.top() == result--);
+        s2.pop();
+    }
+
+    queue q1(range.begin(), range.end());
+    static_assert(is_same_v<decltype(q1), queue<int, deque<int, allocator<int>>>>);
+    assert(q1.size() == size(some_data));
+    result = 0;
+    while (!q1.empty()) {
+        assert(q1.front() == result++);
+        q1.pop();
+    }
+
+    queue q2(range.begin(), range.end(), custom_allocator<int>{});
+    static_assert(is_same_v<decltype(q2), queue<int, deque<int, custom_allocator<int>>>>);
+    assert(q2.size() == size(some_data));
+    result = 0;
+    while (!q2.empty()) {
+        assert(q2.front() == result++);
+        q2.pop();
+    }
+#endif // _HAS_CXX23
+
+#if _HAS_CXX17
+    priority_queue pq1(range.begin(), range.end());
+    static_assert(is_same_v<decltype(pq1), priority_queue<int, vector<int>, less<int>>>);
+#else // ^^^_HAS_CXX17 ^^^ / vvv !_HAS_CXX17 vvv
+    priority_queue<int, vector<int>, less<int>> pq1(range.begin(), range.end());
+#endif // !_HAS_CXX17
+    assert(pq1.size() == size(some_data));
+    result = 5;
+    while (!pq1.empty()) {
+        assert(pq1.top() == result--);
+        pq1.pop();
+    }
+
+#if _HAS_CXX17
+    priority_queue pq2(range.begin(), range.end(), custom_allocator<int>{});
+    static_assert(is_same_v<decltype(pq2), priority_queue<int, vector<int, custom_allocator<int>>, less<int>>>);
+#else // ^^^_HAS_CXX17 ^^^ / vvv !_HAS_CXX17 vvv
+    priority_queue<int, vector<int, custom_allocator<int>>, less<int>> pq2(
+        range.begin(), range.end(), custom_allocator<int>{});
+#endif // !_HAS_CXX17
+    assert(pq2.size() == size(some_data));
+    result = 5;
+    while (!pq2.empty()) {
+        assert(pq2.top() == result--);
+        pq2.pop();
+    }
+
+#if _HAS_CXX17
+    priority_queue pq3(range.begin(), range.end(), greater<int>{}, custom_allocator<int>{});
+    static_assert(is_same_v<decltype(pq3), priority_queue<int, vector<int, custom_allocator<int>>, greater<int>>>);
+#else // ^^^_HAS_CXX17 ^^^ / vvv !_HAS_CXX17 vvv
+    priority_queue<int, vector<int, custom_allocator<int>>, greater<int>> pq3(
+        range.begin(), range.end(), greater<int>{}, custom_allocator<int>{});
+#endif // !_HAS_CXX17
+    assert(pq3.size() == size(some_data));
+    result = 0;
+    while (!pq3.empty()) {
+        assert(pq3.top() == result++);
+        pq3.pop();
+    }
+
+    vector<int> cont(begin(additional_data), end(additional_data));

```

Comment:
`priority_queue` defaults to `vector`, so this doesn't verify that custom container types are correctly picked up. Using `deque` would provide such verification.